services/webnn/ort/graph_builder_ort.cc

```diff
@@ -891,6 +923,60 @@ void GraphBuilderOrt::AddReshapeOperation(const mojom::Reshape& reshape) {
   model_builder_.AddNode(kOpTypeReshape, node_name, input_names, output_names);
 }
 
+void GraphBuilderOrt::AddSliceOperation(const mojom::Slice& slice) {
+  const std::string node_name = GetNodeName(slice.label);
+  const std::string input_name = GetOperandName(slice.input_operand_id);
+  const std::string output_name = GetOperandName(slice.output_operand_id);
+
+  auto range = slice.ranges;
+  base::FixedArray<int64_t> beginnings(slice.ranges.size());
+  base::FixedArray<int64_t> endings(slice.ranges.size());
+  base::FixedArray<int64_t> strides(slice.ranges.size());
+  for (size_t i = 0; i < slice.ranges.size(); ++i) {
+    beginnings[i] = base::checked_cast<int64_t>(slice.ranges[i].start);
+    endings[i] = base::checked_cast<int64_t>(slice.ranges[i].start +
+                                             slice.ranges[i].size);
+    strides[i] = base::checked_cast<int64_t>(slice.ranges[i].stride);
+  }
+
+  // Starts is an operand with data type int64, not an attribute.
+  std::vector<uint32_t> starts_shape = {
+      base::checked_cast<uint32_t>(beginnings.size())};
+  // Slice op needs parameter *starts* as raw data to do shape inference.
+  const std::string starts_name = CreateInitializerAsRawData(
+      starts_shape,
+      base::span(reinterpret_cast<const uint8_t*>(beginnings.data()),
+                 sizeof(int64_t) * beginnings.size()),
+      OperandDataType::kInt64);
+
+  // Ends is an operand with data type int64, not an attribute.
+  std::vector<uint32_t> ends_shape = {
+      base::checked_cast<uint32_t>(endings.size())};
+  // Slice op needs parameter *ends* as raw data to do shape inference.
+  const std::string ends_name = CreateInitializerAsRawData(
+      ends_shape,
+      base::span(reinterpret_cast<const uint8_t*>(endings.data()),
+                 sizeof(int64_t) * endings.size()),
+      OperandDataType::kInt64);
+
+  // Steps is an operand with data type int64, not an attribute.
+  std::vector<uint32_t> steps_shape = {
+      base::checked_cast<uint32_t>(strides.size())};
+  // Slice op needs parameter *steps* as raw data to do shape inference.
+  const std::string steps_name = CreateInitializerAsRawData(
+      steps_shape,
+      base::span(reinterpret_cast<const uint8_t*>(strides.data()),
+                 sizeof(int64_t) * strides.size()),
+      OperandDataType::kInt64);
+
+  std::array<const char*, 5> input_names = {
+      input_name.c_str(), starts_name.c_str(), ends_name.c_str(), /*axes=*/"",
```

Comment:
Worth adding a comment explaining what's the behavior when axes is not present

https://onnx.ai/onnx/operators/onnx__Slice.html

> If axes are omitted, they are set to [0, ..., r-1]

Reply:
Done and please take another look, thanks a lot!